boa/src/exec/mod.rs

```diff
@@ -200,6 +200,65 @@ impl Interpreter {
         }
     }
 
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_index(&mut self, value: &Value) -> Result<usize, Value> {
+        if value.is_undefined() {
+            return Ok(0);
+        }
+
+        let integer_index = self.to_integer(value)?;
+
+        if integer_index < 0 {
+            self.throw_range_error("Integer index must be >= 0")?;
+            unreachable!();
+        }
+
+        if integer_index > 2i64.pow(53) - 1 {
+            self.throw_range_error("Integer index must be less than 2**(53) - 1")?;
+            unreachable!()
+        }
+
+        Ok(integer_index as usize)
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_integer(&mut self, value: &Value) -> Result<i64, Value> {
+        let number = self.to_number(value)?;
+
+        if number.is_nan() {
+            return Ok(0);
+        }
+
+        Ok(number as i64)
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_number(&mut self, value: &Value) -> Result<f64, Value> {
+        match *value.deref().borrow() {
+            ValueData::Null => Ok(0.0),
+            ValueData::Undefined => Ok(f64::NAN),
+            ValueData::Boolean(b) => Ok(if b { 1.0 } else { 0.0 }),
+            ValueData::String(ref string) => match string.parse::<f64>() {
+                Ok(number) => Ok(number),
+                Err(_) => Ok(0.0),
+            }, // this is probably not 100% correct, see https://tc39.es/ecma262/#sec-tonumber-applied-to-the-string-type
+            ValueData::Rational(number) => Ok(number),
+            ValueData::Integer(integer) => Ok(integer as f64),
```

Comment:
```suggestion
            ValueData::Integer(integer) => Ok(f64::from(*integer)),
```
It's better if we use from conversion rather than `as` conversions, the `i32` can fit in `f64`